generate file
* snps_OL13_rsid_list_unmapped.txt

In [15]:
suppressMessages(suppressWarnings(source("../config/config_sing.R")))
show_env()

You are in Singularity: singularity_proj_combeffect 
BASE DIRECTORY:     /data/reddylab/Kuei 
WORK DIRECTORY:     /data/reddylab/Kuei/out 
CODE DIRECTORY:     /data/reddylab/Kuei/code 
PATH OF SOURCE:     /data/reddylab/Kuei/source 
PATH OF EXECUTABLE: /data/reddylab/Kuei/bin 
PATH OF ANNOTATION: /data/reddylab/Kuei/annotation 
PATH OF PROJECT:    /data/reddylab/Kuei/code/Proj_CombEffect_ENCODE_FCC 
PATH OF RESULTS:    /data/reddylab/Kuei/out/proj_combeffect_encode_fcc 


## Import data (SNPS)

In [16]:
###
fdiry = file.path(FD_RES, "region", "control_tmpra")
fname = "snps_OL13_rsid_info.tsv"
fpath = file.path(fdiry, fname)

###
dat = read_tsv(fpath)

###
dat_snps_OL13_rsid_info = dat
print(dim(dat))
head(dat)

Rows: 198 Columns: 3
── Column specification ───────────────────────────────────────────────────────────
Delimiter: "\t"
chr (3): Dataset, Project, SNP

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] 198   3


Dataset Project   SNP       
1 OL13    negCtrl   rs10035423
2 OL13    emVarCtrl rs1008917 
3 OL13    negCtrl   rs10146602
4 OL13    expCtrl   rs10146759
5 OL13    emVarCtrl rs1016110 
6 OL13    negCtrl   rs10201513

## Import query results

In [17]:
### set file directory
fdiry = file.path(FD_RES, "region", "control_tmpra")
fname = "snps_OL13_rsid_query_kaviar_hg38.vcf"
fpath = file.path(fdiry, fname)

### import data
dat = read_tsv(fpath, comment = "##")

### show info
dat_snps_OL13_rsid_query_hg38 = dat
print(dim(dat))
head(dat)

Warning message:
“One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat <- vroom(...)
  problems(dat)”
Rows: 274 Columns: 8
── Column specification ───────────────────────────────────────────────────────────
Delimiter: "\t"
chr (7): #CHROM, ID, REF, ALT, QUAL, FILTER, INFO
dbl (1): POS

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] 274   8


#CHROM POS      ID          REF  ALT  QUAL FILTER
1 1      10391742 rs4846217   C    T    .    .     
2 1      28364347 rs4908346   C    T    .    .     
3 1      28364347 rs796493340 CA   TG   .    .     
4 1      28364347 .           CAAA TGAA .    .     
5 1      36523011 rs12060242  G    C    .    .     
6 1      52838378 rs115675756 G    A    .    .     
  INFO                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
1 AF=0.1127076;AC=2973;AN=26378;DS=GS000010327|GS000010426|GS000010429|GS000010430|GS000010432|GS000010434|GS000011737|GS000011805|GS000011810|GS000011811|GS000011816|GS000014558|GS000014560|GS000014561|GS000014566|GS000015154|GS000015172|GS000015177|GS000015181|GS000015185|GS000015187|GS000015221|GS000015225|GS000015228|GS000015231|GS000015232|GS000015233|GS000015891|GS000015894|GS000016016|GS000016027|GS000016336|GS000016337|GS000016339|GS000016375|GS000016441|GS000016446|GS000016448|HGDP00521|HGDP00665|HGDP01284|ISB_founders-Nge3|Inova_CGI_founders-Nge3|Inova_Illumina_founders-Nge3|Malay|NA12889|NA21732|SS6004471|SSIP|UK10K|Wellderly|gonl|phase3-ACB|phase3-ASW|phase3-BEB|phase3-CDX|phase3-CEU|phase3-CHB|phase3-CLM|phase3-ESN|phase3-FIN|phase3-GBR|phase3-GIH|phase3-GWD|phase3-IBS|phase3-ITU|phase3-JPT|phase3-KHV|phase3-LWK|phase3-MSL|phase3-MXL|phase3-PEL|phase3-PJL|phase3-PUR|phase3-STU|phase3-TSI|phase3-YRI                                                                                                                   
2 AF=0.1932671;AC=5098;AN=26378;DS=DNK02|GMIAK4|HGDP00521|HGDP00665|HGDP00927|HGDP0456|Inova_Illumina_founders-Nge3|Malay|SAIF|SS6004467|SS6004470|SS6004471|SS6004473|SS6004474|SS6004475|SSIP|Saqqaq|UK10K|gonl|phase3-ACB|phase3-ASW|phase3-BEB|phase3-CDX|phase3-CEU|phase3-CHB|phase3-CHS|phase3-CLM|phase3-ESN|phase3-FIN|phase3-GBR|phase3-GIH|phase3-GWD|phase3-IBS|phase3-ITU|phase3-JPT|phase3-KHV|phase3-LWK|phase3-MSL|phase3-MXL|phase3-PEL|phase3-PJL|phase3-PUR|phase3-STU|phase3-TSI|phase3-YRI                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
3 AF=0.0581166;AC=1533;AN=26378;END=28364348;DS=GS000009920|GS000009928|GS000009930|GS000009931|GS000009932|GS000010320|GS000010323|GS000010327|GS000010427|GS000010428|GS000010432|GS000010435|GS000010437|GS000010494|GS000011737|GS000011739|GS000011740|GS000011741|GS000011804|GS000011805|GS000011807|GS000011809|GS000011812|GS000011814|GS000011815|GS000011816|GS000012712|GS000014558|GS000014559|GS000014560|GS000014561|GS000014562|GS000014564|GS000014565|GS000014570|GS000015152|GS000015172|GS000015174|GS000015176|GS

In [18]:
problems(dat)

row col expected  actual    file
1  73 1   8 columns 1 columns     
2 100 1   8 columns 1 columns     
3 196 1   8 columns 1 columns     
4 247 1   8 columns 1 columns     
5 248 1   8 columns 1 columns

In [19]:
### set file directory
fdiry = file.path(FD_RES, "region", "control_tmpra")
fname = "snps_OL13_rsid_query_kaviar_hg19.vcf"
fpath = file.path(fdiry, fname)

### import data
dat = read_tsv(fpath, comment = "##")

### show info
dat_snps_OL13_rsid_query_hg19 = dat
print(dim(dat))
head(dat)

Warning message:
“One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat <- vroom(...)
  problems(dat)”
Rows: 274 Columns: 8
── Column specification ───────────────────────────────────────────────────────────
Delimiter: "\t"
chr (7): #CHROM, ID, REF, ALT, QUAL, FILTER, INFO
dbl (1): POS

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] 274   8


#CHROM POS      ID          REF  ALT  QUAL FILTER
1 1      10451799 rs4846217   C    T    .    .     
2 1      28690858 rs4908346   C    T    .    .     
3 1      28690858 rs796493340 CA   TG   .    .     
4 1      28690858 .           CAAA TGAA .    .     
5 1      36988612 rs12060242  G    C    .    .     
6 1      53304050 rs115675756 G    A    .    .     
  INFO                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
1 AF=0.1127076;AC=2973;AN=26378;DS=GS000010327|GS000010426|GS000010429|GS000010430|GS000010432|GS000010434|GS000011737|GS000011805|GS000011810|GS000011811|GS000011816|GS000014558|GS000014560|GS000014561|GS000014566|GS000015154|GS000015172|GS000015177|GS000015181|GS000015185|GS000015187|GS000015221|GS000015225|GS000015228|GS000015231|GS000015232|GS000015233|GS000015891|GS000015894|GS000016016|GS000016027|GS000016336|GS000016337|GS000016339|GS000016375|GS000016441|GS000016446|GS000016448|HGDP00521|HGDP00665|HGDP01284|ISB_founders-Nge3|Inova_CGI_founders-Nge3|Inova_Illumina_founders-Nge3|Malay|NA12889|NA21732|SS6004471|SSIP|UK10K|Wellderly|gonl|phase3-ACB|phase3-ASW|phase3-BEB|phase3-CDX|phase3-CEU|phase3-CHB|phase3-CLM|phase3-ESN|phase3-FIN|phase3-GBR|phase3-GIH|phase3-GWD|phase3-IBS|phase3-ITU|phase3-JPT|phase3-KHV|phase3-LWK|phase3-MSL|phase3-MXL|phase3-PEL|phase3-PJL|phase3-PUR|phase3-STU|phase3-TSI|phase3-YRI                                                                                                                   
2 AF=0.1932671;AC=5098;AN=26378;DS=DNK02|GMIAK4|HGDP00521|HGDP00665|HGDP00927|HGDP0456|Inova_Illumina_founders-Nge3|Malay|SAIF|SS6004467|SS6004470|SS6004471|SS6004473|SS6004474|SS6004475|SSIP|Saqqaq|UK10K|gonl|phase3-ACB|phase3-ASW|phase3-BEB|phase3-CDX|phase3-CEU|phase3-CHB|phase3-CHS|phase3-CLM|phase3-ESN|phase3-FIN|phase3-GBR|phase3-GIH|phase3-GWD|phase3-IBS|phase3-ITU|phase3-JPT|phase3-KHV|phase3-LWK|phase3-MSL|phase3-MXL|phase3-PEL|phase3-PJL|phase3-PUR|phase3-STU|phase3-TSI|phase3-YRI                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
3 AF=0.0581166;AC=1533;AN=26378;END=28690859;DS=GS000009920|GS000009928|GS000009930|GS000009931|GS000009932|GS000010320|GS000010323|GS000010327|GS000010427|GS000010428|GS000010432|GS000010435|GS000010437|GS000010494|GS000011737|GS000011739|GS000011740|GS000011741|GS000011804|GS000011805|GS000011807|GS000011809|GS000011812|GS000011814|GS000011815|GS000011816|GS000012712|GS000014558|GS000014559|GS000014560|GS000014561|GS000014562|GS000014564|GS000014565|GS000014570|GS000015152|GS000015172|GS000015174|GS000015176|GS

In [20]:
problems(dat)

row col expected  actual    file
1  73 1   8 columns 1 columns     
2 100 1   8 columns 1 columns     
3 196 1   8 columns 1 columns     
4 247 1   8 columns 1 columns     
5 248 1   8 columns 1 columns

## Merge

In [22]:
### init
lst = list(
    "hg38" = dat_snps_OL13_rsid_query_hg38,
    "hg19" = dat_snps_OL13_rsid_query_hg19
)

### process
lst = lapply(lst, function(dat){
    dat = dat %>%
        dplyr::mutate(
            RSID     = ID,
            Chrom    = paste0("chr", `#CHROM`),
            Loc = POS,
            Allele_Ref = REF,
            Allele_Alt = ALT
        ) %>%
        dplyr::filter(str_detect(RSID, "rs")) %>%
        dplyr::select(RSID, Chrom, Loc, Allele_Ref, Allele_Alt) 
    return(dat)
})

### combine and arrange
dat = bind_rows(lst, .id = "Genome")
dat = dat %>% 
    dplyr::mutate(Genome = paste("Loc", Genome, sep="_")) %>% 
    dplyr::distinct() %>%
    tidyr::spread(Genome, Loc) 

chroms = paste0("chr", c(1:22, "X", "Y"))
dat$Chrom = factor(dat$Chrom, levels=chroms)
dat = dat %>% dplyr::arrange(Chrom, Loc_hg19)

### assign and show
dat_snps_OL13_rsid_query = dat
head(dat)

RSID        Chrom Allele_Ref Allele_Alt Loc_hg19 Loc_hg38
1 rs4846217   chr1  C          T          10451799 10391742
2 rs4908346   chr1  C          T          28690858 28364347
3 rs796493340 chr1  CA         TG         28690858 28364347
4 rs12060242  chr1  G          C          36988612 36523011
5 rs115675756 chr1  G          A          53304050 52838378
6 rs11163961  chr1  C          T          76151253 75685568

## Check if the SNPs has the correct information

In [24]:
set.seed(123)
dat = dat_snps_OL13_rsid_query
dat = slice_sample(dat, n = 20) %>% arrange(RSID)
dat

RSID        Chrom Allele_Ref Allele_Alt Loc_hg19  Loc_hg38 
1  rs1016110   chr22 A          C           42836337  42440331
2  rs10490679  chr2  A          G           36497661  36270518
3  rs114035764 chr19 C          T           52684996  52181743
4  rs139126    chr22 T          A           44577160  44181280
5  rs149201729 chr21 G          A           32350413  30978094
6  rs1566698   chr9  A          G           91480139  88865224
7  rs2216063   chr16 G          A           54387084  54353172
8  rs2384273   chr10 C          G           35272662  34983734
9  rs2850402   chr4  T          C          108065576 107144419
10 rs2854183   chr17 C          T           61968002  63890642
11 rs386715665 chr7  TGT        AGC,AGT     89618632  89989318
12 rs4546732   chr9  A          C           81584066  78969150
13 rs4855158   chr3  T          C           86717820  86668670
14 rs62108166  chr19 C          T           13907805  13796991
15 rs62356690  chr5  C          A           58387311  59091484
16 rs7116100   chr11 T          C          111211739 111341014
17 rs722643    chr14 C          T           92641615  92175271
18 rs7550949   chr1  C          G           76166846  75701161
19 rs76819181  chr9  G          T           92464255  89762808
20 rs9749442   chr19 C          G           58144519  57633151

- rs1016110    https://www.ncbi.nlm.nih.gov/snp/rs1016110
    - chr22:42440331 (GRCh38.p14)
    - A>C / A>G
- rs10490679   https://www.ncbi.nlm.nih.gov/snp/rs10490679
- rs114035764  https://www.ncbi.nlm.nih.gov/snp/rs114035764
- rs139126     https://www.ncbi.nlm.nih.gov/snp/rs139126
- rs149201729  https://www.ncbi.nlm.nih.gov/snp/rs149201729
- rs1566698    https://www.ncbi.nlm.nih.gov/snp/rs1566698
- rs2216063    https://www.ncbi.nlm.nih.gov/snp/rs2216063
- rs2384273    https://www.ncbi.nlm.nih.gov/snp/rs2384273
- rs2850402    https://www.ncbi.nlm.nih.gov/snp/rs2850402
- rs2854183    https://www.ncbi.nlm.nih.gov/snp/rs2854183
- rs386715665  https://www.ncbi.nlm.nih.gov/snp/rs386715665
- rs4546732    https://www.ncbi.nlm.nih.gov/snp/rs4546732
- rs4855158    https://www.ncbi.nlm.nih.gov/snp/rs4855158
- rs62108166   https://www.ncbi.nlm.nih.gov/snp/rs62108166
- rs62356690   https://www.ncbi.nlm.nih.gov/snp/rs62356690
- rs7116100    https://www.ncbi.nlm.nih.gov/snp/rs7116100
- rs722643     https://www.ncbi.nlm.nih.gov/snp/rs722643
- rs7550949    https://www.ncbi.nlm.nih.gov/snp/rs7550949
- rs76819181   https://www.ncbi.nlm.nih.gov/snp/rs76819181
- rs9749442    https://www.ncbi.nlm.nih.gov/snp/rs9749442

## Figure out the unmapped SNPs

In [26]:
###
dat = dat_snps_OL13_rsid_info

###
dat = dat %>% 
    dplyr::select(SNP) %>%
    dplyr::left_join(dat_snps_OL13_rsid_query, by=c("SNP"="RSID"))

###
dat_snps_combine = dat
print(dim(dat))
head(dat)

[1] 199   6


SNP        Chrom Allele_Ref Allele_Alt Loc_hg19  Loc_hg38 
1 rs10035423 chr5  A          G          119306380 119970685
2 rs1008917  chr15 T          C           65059651  64767452
3 rs10146602 chr14 T          C           89693105  89226761
4 rs10146759 chr14 C          T           24899194  24429988
5 rs1016110  chr22 A          C           42836337  42440331
6 rs10201513 chr2  A          C           88955614  88656096

In [27]:
dat = dat_snps_combine
dat = dat %>% dplyr::filter(is.na(Chrom))

dat_snps_ummapped = dat
print(dim(dat))
head(dat)

[1] 14  6


SNP         Chrom Allele_Ref Allele_Alt Loc_hg19 Loc_hg38
1 rs112554467 NA    NA         NA         NA       NA      
2 rs113065267 NA    NA         NA         NA       NA      
3 rs115014431 NA    NA         NA         NA       NA      
4 rs116025516 NA    NA         NA         NA       NA      
5 rs116100242 NA    NA         NA         NA       NA      
6 rs117188042 NA    NA         NA         NA       NA

## Store th unmapped SNPs into a list

In [28]:
fdiry = file.path(FD_RES, "region", "control_tmpra")
fname = "snps_OL13_rsid_list_unmapped.tsv"
fpath = file.path(fdiry, fname)

dat = dat_snps_ummapped 
write_tsv(dat, fpath)